dox/CMakeLists.txt

```diff
@@ -15,6 +15,32 @@ if (NOT PYTHONINTERP_FOUND)
     return ()
 endif (NOT PYTHONINTERP_FOUND)
 
+# find python 2.x
+if(NOT(${PYTHON_VERSION_MAJOR} MATCHES "2"))
+    set(_PYTHON2_VERSIONS 2.7 2.6 2.5 2.4 2.3 2.2 2.1 2.0)
+    unset(_Python_NAMES)
+    foreach(_CURRENT_VERSION IN LISTS _PYTHON2_VERSIONS)
+        list(APPEND _Python_NAMES "python${_CURRENT_VERSION}")
+    endforeach()
+    unset(_PYTHON2_VERSIONS)
+
+    find_program(PYTHON2_EXECUTABLE NAMES ${_Python_NAMES})
+    if(PYTHON2_EXECUTABLE MATCHES "PYTHON2_EXECUTABLE-NOTFOUND")
+        message (STATUS "You need Python 2.x for building dox. (skip the tests)")
+        return()
+    endif()
+    set(PYTHON_EXECUTABLE "${PYTHON2_EXECUTABLE}")
+    message (STATUS "Found Python 2.x: ${PYTHON_EXECUTABLE}")
+
+    # change shell scripts for building dox
+    file(READ "${CMAKE_CURRENT_SOURCE_DIR}/../dox/run.sh" run_content)
+    file(READ "${CMAKE_CURRENT_SOURCE_DIR}/../dox/dox_only.sh" dox_only_content)
+    string(REPLACE "python2 ../" "${PYTHON_EXECUTABLE} ../" run_content "${run_content}")
+    string(REPLACE "python2 ../" "${PYTHON_EXECUTABLE} ../" dox_only_content "${dox_only_content}")
+    file(WRITE "${CMAKE_CURRENT_SOURCE_DIR}/../dox/run.sh" "${run_content}")
+    file(WRITE "${CMAKE_CURRENT_SOURCE_DIR}/../dox/dox_only.sh" "${dox_only_content}")
```

Comment:
??? No it won't. You can use a special env like ```SEQAN_PYTHON_INTERPRETER```, which is unlikely to interfere with the user's env settings. This is only set by the cmake script. In the shell you can now set: 
```set py_interpreter="${SEQAN_PYTHON_INTERPRETER}-python2}"```
So the behavior is exactly the same as when reading the script replacing the value and writing it out again. But in case of our build system we have set the env and use the correct py version. Or do I miss something?